tracer/src/Datadog.Trace/Util/UriHelpers.cs

```diff
@@ -24,12 +24,20 @@ public static string CleanUri(Uri uri, bool removeScheme, bool tryRemoveIds)
             {
                 // keep only host and path.
                 // remove scheme, userinfo, query, and fragment.
+#if NET6_0_OR_GREATER
+                return string.Create(null, stackalloc char[512 + path.Length], $"{uri.Authority}{path}");
```

Comment:
My gut says `512 + path.Length` is too large for a `stackalloc`. char takes up 2 bytes so this seems risky. I think the most I've seen in dotnet/runtime is `stackalloc char[128]` 🤷

@stephentoub Any light you can shed on this? Sorry for the random mention your work inspired this PR 😄

Reply:
Yes, I will keep only small constant values.